backend/module/chaos/serverexperimentation/xds/ecds.go

```diff
@@ -0,0 +1,255 @@
+package xds
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	gcpCoreV3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	gcpRoute "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+	gcpFilterCommon "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/common/fault/v3"
+	gcpFilterFault "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/fault/v3"
+	gcpType "github.com/envoyproxy/go-control-plane/envoy/type/v3"
+	gcpTypes "github.com/envoyproxy/go-control-plane/pkg/cache/types"
+	proto "github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/golang/protobuf/ptypes/duration"
+	"go.uber.org/zap"
+
+	experimentation "github.com/lyft/clutch/backend/api/chaos/experimentation/v1"
+	serverexperimentation "github.com/lyft/clutch/backend/api/chaos/serverexperimentation/v1"
+)
+
+const (
+	EnvoyDownstreamServiceClusterHeader   = `x-envoy-downstream-service-cluster`
+	FaultFilterConfigNameForExternalFault = `envoy.egress.extension_config.%s`
+	FaultFilterConfigNameForInternalFault = `envoy.extension_config`
```

Comment:
These can all be private I would think? same with a lot of the other variables in this file I believe